File: backups/import_rewrite/core/models/notification.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Text, DateTime, Boolean, ForeignKey, Integer, Float
from backend.db.base import Base


class Notification(Base):
    __tablename__ = "notifications"
    id = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String, ForeignKey("users.id"), index=True, nullable=False)
    type = Column(String, default="info")  # info/warning/error/success
    title = Column(String, nullable=False)
    message = Column(Text, nullable=False)
    is_read = Column(Boolean, default=False)
    metadata_json = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)

File: frontend/src/types/notification.ts
export interface NotificationMetadata {
  [key: string]: unknown;
}

export interface Notification {
  id: string;

  user_id: string;

  type: string;

  title: string;

  message: string;

  is_read: boolean;

  metadata: NotificationMetadata | null;

  created_at: string | null;
}

export interface NotificationResponse {
  total: number;

  skip: number;

  limit: number;

  items: Notification[];
}